Realm/Realm/DatabaseTypes/RealmCollectionBase.cs

```diff
@@ -396,18 +423,31 @@ private void UpdateCollectionChangedSubscriptionIfNecessary(bool isSubscribed)
             {
                 if (isSubscribed)
                 {
-                    SubscribeForNotificationsImpl(OnChange, shallow: true);
+                    SubscribeForNotificationsImpl(OnChange, KeyPathsCollection.Shallow);
                 }
                 else
                 {
-                    UnsubscribeFromNotifications(OnChange, shallow: true);
+                    UnsubscribeFromNotifications(OnChange, KeyPathsCollectionType.Shallow);
                 }
             }
         }
 
         #endregion INotifyCollectionChanged
 
-        void INotifiable<NotifiableObjectHandleBase.CollectionChangeSet>.NotifyCallbacks(NotifiableObjectHandleBase.CollectionChangeSet? changes, bool shallow)
+        void INotifiable<CollectionChangeSet>.NotifyCallbacks(CollectionChangeSet? changes, KeyPathsCollectionType type, IntPtr callbackNative)
+        {
+            if (type == KeyPathsCollectionType.Full
+                && GCHandle.FromIntPtr(callbackNative).Target is NotificationCallbackDelegate<T> callback)
+            {
+                callback(this, GetFromNative(changes));
+                return;
+            }
+
+            var changeset = GetFromNative(changes);
+            _notificationCallbacks.Value.Notify(changeset, type);
+        }
+
+        private ChangeSet? GetFromNative(CollectionChangeSet? changes)
```

Comment:
I don't think this needs a separate method - we still need the changeset only in `NotifyCallbacks` and regardless of what the type is.